library/spdm_common_lib/libspdm_com_opaque_data.c

```diff
@@ -230,3 +230,93 @@ bool libspdm_get_element_from_opaque_data(libspdm_context_t *spdm_context,
+ * @param  data_in                       A pointer to the buffer to store the opaque data version selection.
+ *
+ * @retval true                           check opaque data successfully
+ * @retval false                          check opaque data failed
+ **/
+bool libspdm_process_general_opaque_data_check(libspdm_context_t *spdm_context,
+                                               size_t data_in_size,
+                                               void *data_in)
+{
+    const spdm_general_opaque_data_table_header_t
+    *spdm_general_opaque_data_table_header;
+    const opaque_element_table_header_t
+    *opaque_element_table_header;
+    uint8_t element_num;
+    uint8_t element_index;
+    uint16_t opaque_element_data_len;
+    size_t data_element_size;
+    size_t current_element_len;
+    size_t total_element_len;
+
+    total_element_len = 0;
+
+    LIBSPDM_ASSERT(data_in_size <= SPDM_MAX_OPAQUE_DATA_SIZE);
+
+    if (libspdm_get_connection_version(spdm_context) >= SPDM_MESSAGE_VERSION_12) {
+        if (spdm_context->connection_info.algorithm.other_params_support ==
+            SPDM_ALGORITHMS_OPAQUE_DATA_FORMAT_1) {
+            /* Check byte alignment */
+            if ((data_in_size & 3) != 0) {
+                return false;
+            }
+
+            spdm_general_opaque_data_table_header = data_in;
+            if (data_in_size < sizeof(spdm_general_opaque_data_table_header_t)) {
+                return false;
+            }
+            if (spdm_general_opaque_data_table_header->total_elements < 1) {
+                return false;
+            }
+            opaque_element_table_header = (const void *)(spdm_general_opaque_data_table_header + 1);
+
+            element_num = spdm_general_opaque_data_table_header->total_elements;
+
+            data_element_size = data_in_size - sizeof(spdm_general_opaque_data_table_header_t);
+
+            for (element_index = 0; element_index < element_num; element_index++) {
+                /*ensure the opaque_element_table_header is valid*/
+                if (total_element_len + sizeof(opaque_element_table_header_t) +
+                    sizeof(opaque_element_data_len) >
+                    data_element_size) {
+                    return false;
+                }
+
+                /*check element header id*/
+                if (opaque_element_table_header->id > SPDM_REGISTRY_ID_JEDEC) {
+                    return false;
+                }
+
+                opaque_element_data_len = *(uint16_t *)((size_t)(opaque_element_table_header + 1)) +
+                                          opaque_element_table_header->vendor_len;
+
+                current_element_len = sizeof(opaque_element_table_header_t) +
+                                      opaque_element_table_header->vendor_len +
+                                      sizeof(opaque_element_data_len) +
+                                      opaque_element_data_len;
+                /* Add Padding*/
```

Comment:
Can also check that if there is padding then the padding bytes are zeroes.

Reply:
Thanks. Padding byte check updated